devU-api/src/containerAutoGrader/containerAutoGrader.validator.ts

```diff
@@ -0,0 +1,23 @@
+import { check } from 'express-validator'
+
+import validate from '../middleware/validator/generic.validator'
+
+
+
+const assignmentId = check('assignmentId').isNumeric()
+
+const graderFilename = check('graderFilename')
+.custom((_value, { req }) => req.file.size > 0)
+.withMessage('Grader file is required')
+
+const makefileFilename = check('makefileFilename').isString().trim().optional({ nullable: true })
```

Comment:
This should be handled the same way as grader files, but with optional

Reply:
solved via commit id [https://github.com/makeopensource/devU/commit/a05bf000a13a38c94aa7bf58f6aea666deb53b4b], validator will now check if the input makefile is null or content with size.